DOCSP-27376: removed support for dot notation in find* (#491)
mongodb/docs-node: source/whats-new.txt

New features of the 5.0 {+driver-short+} release include:

> - By default, the driver no longer checks types referenced in dot notation
>   unless the ``StrictFilter`` feature is explicitly
>   implemented. To learn more about this change, see the :ref:`Typescript fundamentals page <node-ts-type-safety>`.

Reviewer: ```suggestion
- By default, the driver no longer checks types referenced in dot notation
  unless the ``StrictFilter`` type annotation is explicitly
  used. To learn more about this change, see the :ref:`Typescript fundamentals page <node-ts-type-safety>`.
```

Reviewer: Additionally, in our migration guide we made a **bold header call out** that this is a TypeScript only difference that does not effect runtime database queries/operations. I'm sure you have your own style rules but we felt it's important that there isn't confusion that our driver does special handling of keys with dots in them at runtime.

- Optional installation of ``@aws-sdk/credential-providers`` as a dependency.

>   - The driver no longer includes AWS SDK modules by default. Use the

Reviewer: Sorry this is unrelated to the work here, the aws sdk version below isn't correct, and some shells don't work well with `@` and caret symbols `^` (which we should also add to the version) so quoting it is a good idea for maximum compat:

```bash
npm install --save "@aws-sdk/credential-providers@^3.201.0"
```

New features of the 5.0 {+driver-short+} release include:

> - Removal of default Typescript type-checking of types expressed in dot
>   notation.
> 
>   - The driver no longer checks types referenced in dot notation during
>     find operations unless the ``StrictFilter`` feature is explicitly
>     implemented. To learn more about this change, see the :ref:`Typescript fundamentals page <node-ts-type-safety>`.

Reviewer 2: ```suggestion
  - By default, the driver no longer checks types referenced in dot notation during
    find operations unless the ``StrictFilter`` feature is explicitly
    implemented. To learn more about this change, see the :ref:`Typescript fundamentals page <node-ts-type-safety>`.
```
